src/shapes/SurfaceShape.js

```diff
@@ -947,43 +949,47 @@ define([
          */
         SurfaceShape.prototype.renderToTexture = function (dc, ctx2D, xScale, yScale, dx, dy) {
             var attributes = (this._highlighted ? (this._highlightAttributes || this._attributes) : this._attributes);
+            if (!attributes) {
+                return;
+            }
+
             var drawInterior = (!this._isInteriorInhibited && attributes.drawInterior);
             var drawOutline = (attributes.drawOutline && attributes.outlineWidth > 0);
-
             if (!drawInterior && !drawOutline) {
                 return;
             }
 
-            if (dc.pickingMode && !this.pickColor) {
-                this.pickColor = dc.uniquePickColor();
-            }
-
             if (dc.pickingMode) {
-                var pickColor = this.pickColor.toHexString();
+                if (!this.pickColor) {
+                    this.pickColor = dc.uniquePickColor();
+                }
+                ctx2D.fillStyle = this.pickColor.toHexString();
```

Comment:
@zglueck Although the default toHexString behaviour probably works fine here (I did not research it), I think it would be better to explicitly specify the value for the isUsingAlpha parameter. It seems more deterministic that way. It looks like the old code had this problem as well.